spell-template/spell.sol

```diff
@@ -79,69 +59,62 @@ contract SpellAction {
-        require(GemJoinAbstract(MCD_JOIN_NS2DRP_A).gem() == NS2DRP_GEM, "join-gem-not-match");
-        require(GemJoinAbstract(MCD_JOIN_NS2DRP_A).dec() == DSTokenAbstract(NS2DRP_GEM).decimals(), "join-dec-not-match");
+        require(GemJoinAbstract(MCD_JOIN).vat() == MCD_VAT, "join-vat-not-match");
+        require(GemJoinAbstract(MCD_JOIN).ilk() == ilk, "join-ilk-not-match");
+        require(GemJoinAbstract(MCD_JOIN).gem() == GEM, "join-gem-not-match");
+        require(GemJoinAbstract(MCD_JOIN).dec() == DSTokenAbstract(GEM).decimals(), "join-dec-not-match");
 
         // init the RwaLiquidationOracle
         // doc: "IPFS Hash"
         // tau: 5 minutes
         RwaLiquidationLike(MIP21_LIQUIDATION_ORACLE).init(
-            ilk, NS2DRP_A_INITIAL_PRICE, DOC, 300
+            ilk, INITIAL_PRICE, DOC, 300
         );
         (,address pip,,) = RwaLiquidationLike(MIP21_LIQUIDATION_ORACLE).ilks(ilk);
-        CHANGELOG.setAddress("PIP_NS2DRP", pip);
+        CHANGELOG.setAddress(pipID, pip);
 
-        // Set price feed for NS2DRP
+        // Set price feed for CF4DRP
         SpotAbstract(MCD_SPOT).file(ilk, "pip", pip);
 
-        // Init NS2DRP in Vat
+        // Init CF4DRP in Vat
         VatAbstract(MCD_VAT).init(ilk);
-        // Init NS2DRP in Jug
+        // Init CF4DRP in Jug
         JugAbstract(MCD_JUG).init(ilk);
 
-        // Allow NS2DRP Join to modify Vat registry
-        VatAbstract(MCD_VAT).rely(MCD_JOIN_NS2DRP_A);
-
-        // Allow RwaLiquidationOracle to modify Vat registry
-        // VatAbstract(MCD_VAT).rely(MIP21_LIQUIDATION_ORACLE);
+        // Allow CF4DRP Join to modify Vat registry
+        VatAbstract(MCD_VAT).rely(MCD_JOIN);
 
         // 5 Million debt ceiling
-        VatAbstract(MCD_VAT).file(ilk, "line", NS2DRP_A_INITIAL_DC);
-        VatAbstract(MCD_VAT).file("Line", VatAbstract(MCD_VAT).Line() + NS2DRP_A_INITIAL_DC);
+        VatAbstract(MCD_VAT).file(ilk, "line", DC);
+        VatAbstract(MCD_VAT).file("Line", VatAbstract(MCD_VAT).Line() + DC);
 
         // No dust
         // VatAbstract(MCD_VAT).file(ilk, "dust", 0)
 
-        // 3.6% stability fee
-        JugAbstract(MCD_JUG).file(ilk, "duty", NS2DRP_THREEPOINTSIX_PERCENT_RATE);
+        // set stability fee
+        JugAbstract(MCD_JUG).file(ilk, "duty", RATE);
 
-        // Set the NS2DRP-A min collateralization ratio (e.g. 105% => X = 105)
-        SpotAbstract(MCD_SPOT).file(ilk, "mat", 100 * RAY / 100);
+        // Set the CF4DRP-A min collateralization ratio)
+        SpotAbstract(MCD_SPOT).file(ilk, "mat", MAT);
 
         // poke the spotter to pull in a price
         SpotAbstract(MCD_SPOT).poke(ilk);
 
         // give the urn permissions on the join adapter
-        GemJoinAbstract(MCD_JOIN_NS2DRP_A).rely(NS2DRP_A_URN);
+        GemJoinAbstract(MCD_JOIN).rely(URN);
 
         // set up the urn
-        RwaUrnLike(NS2DRP_A_URN).hope(NS2DRP_OPERATOR);
+        RwaUrnLike(URN).hope(OPERATOR);
```

Comment:
So this is where we should move the changelog bits I mentioned above, but we also have to deal with the ilk registry.  Last time, we didn't have the capability to add an RWA asset to the registry, but this time we do, and should.  The addition should look like:
```solidity
address ILK_REGISTRY = DssExecLib.getChangelogAddress("ILK_REGISTRY");
IlkRegistryAbstract(ILK_REGISTRY).put(
    ilk,
    MDC_JOIN,
    GEM,
    DSTokenAbstract(GEM).decimals(),
    3,
    pip,
    address(0),
    "RWA003-A: Centrifuge: ConsoleFreight",
    "RWA003-A"
);
```

You can see the code for `put()` here:
https://github.com/makerdao/ilk-registry/blob/master/src/IlkRegistry.sol#L385-L425